This file is to explore the Data and learn how many null values and how much usable data is in the Dataset 

In [44]:
import pandas as pd
df = pd.read_csv('dataset.csv')
num_rows = len(df)
print("Number of rows in df:", num_rows)
countnull = df.isnull().sum().sort_values()
print(countnull)

Number of rows in df: 104272
YearStart                          0
YearEnd                            0
LocationAbbr                       0
LocationDesc                       0
Datasource                         0
Class                              0
Topic                              0
Question                           0
Data_Value_Type                    0
TopicID                            0
QuestionID                         0
DataValueTypeID                    0
LocationID                         0
StratificationCategory1            0
Stratification1                    0
StratificationCategoryId1          0
ClassID                            0
StratificationID1                  0
GeoLocation                     1932
Data_Value                     10767
High_Confidence_Limit          10767
Data_Value_Alt                 10767
Low_Confidence_Limit           10767
Sample_Size                    10767
Data_Value_Unit                15400
Race/Ethnicity                 74480
Income   

Lots of null, not that good

Which classes are present in the Data and how many?

In [45]:
unique_values = df['Class'].value_counts()
print(unique_values)

Class
Physical Activity          55720
Obesity / Weight Status    39368
Fruits and Vegetables       9184
Name: count, dtype: int64


Which Questions have been asked in the data and how many?

In [46]:
unique_values = df['Question'].value_counts()
print(unique_values)

Question
Percent of adults aged 18 years and older who have obesity                                                                                                                                                                                               19684
Percent of adults aged 18 years and older who have an overweight classification                                                                                                                                                                          19684
Percent of adults who engage in no leisure-time physical activity                                                                                                                                                                                        19656
Percent of adults who engage in muscle-strengthening activities on 2 or more days a week                                                                                                                                          

how many null values are there in a year, for example 2019

In [47]:
df_year = df[df['YearStart'] == 2019]
num_rows = len(df_year)
print(num_rows)
countnull = df_year.isnull().sum()
print(countnull)

13608
YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
Datasource                        0
Class                             0
Topic                             0
Question                          0
Data_Value_Unit                   0
Data_Value_Type                   0
Data_Value                     1557
Data_Value_Alt                 1557
Data_Value_Footnote_Symbol    12051
Data_Value_Footnote           12051
Low_Confidence_Limit           1557
High_Confidence_Limit          1557
Sample_Size                    1557
Total                         13122
Age(years)                    10692
Education                     11664
Gender                        12636
Income                        10206
Race/Ethnicity                 9720
GeoLocation                     252
ClassID                           0
TopicID                           0
QuestionID                        0
DataValueTypeID       

how many null values are there in a question?

In [48]:
subset_df = df[df['Question'].isin(['Percent of adults aged 18 years and older who have obesity'])]
num_rows = len(subset_df)
print("Number of rows in subset_df:", num_rows)
countnull = subset_df.isnull().sum()
print(countnull)

Number of rows in subset_df: 19684
YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
Datasource                        0
Class                             0
Topic                             0
Question                          0
Data_Value_Unit                3080
Data_Value_Type                   0
Data_Value                     1990
Data_Value_Alt                 1990
Data_Value_Footnote_Symbol    17694
Data_Value_Footnote           17694
Low_Confidence_Limit           1990
High_Confidence_Limit          1990
Sample_Size                    1990
Total                         18981
Age(years)                    15466
Education                     16872
Gender                        18278
Income                        14763
Race/Ethnicity                14060
GeoLocation                     364
ClassID                           0
TopicID                           0
QuestionID                   

How many null values are in the Class Obesity/Weight Status

In [49]:
subset_df = df[df['Class'].isin(['Obesity / Weight Status'])]
num_rows = len(subset_df)
print("Number of rows in subset_df:", num_rows)
countnull = subset_df.isnull().sum()
print(countnull)

Number of rows in subset_df: 39368
YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
Datasource                        0
Class                             0
Topic                             0
Question                          0
Data_Value_Unit                6160
Data_Value_Type                   0
Data_Value                     3980
Data_Value_Alt                 3980
Data_Value_Footnote_Symbol    35388
Data_Value_Footnote           35388
Low_Confidence_Limit           3980
High_Confidence_Limit          3980
Sample_Size                    3980
Total                         37962
Age(years)                    30932
Education                     33744
Gender                        36556
Income                        29526
Race/Ethnicity                28120
GeoLocation                     728
ClassID                           0
TopicID                           0
QuestionID                   

3980 null in DataValue is not good, 3890 in sampel size is also not good, those rows are useless to me and will be deleted. Rows missing Age, Education, Gender and income will have to be reviued by statification classification to find out if they are meant to be missing or unintentionally missing. 

In [50]:
subset_df = df[df['Class'].isin(['Physical Activity'])]
num_rows = len(subset_df)
print("Number of rows in subset_df:", num_rows)
countnull = subset_df.isnull().sum()
print(countnull)

Number of rows in subset_df: 55720
YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
Datasource                        0
Class                             0
Topic                             0
Question                          0
Data_Value_Unit                9240
Data_Value_Type                   0
Data_Value                     5758
Data_Value_Alt                 5758
Data_Value_Footnote_Symbol    49962
Data_Value_Footnote           49962
Low_Confidence_Limit           5758
High_Confidence_Limit          5758
Sample_Size                    5758
Total                         53730
Age(years)                    43780
Education                     47760
Gender                        51740
Income                        41790
Race/Ethnicity                39800
GeoLocation                    1036
ClassID                           0
TopicID                           0
QuestionID                   

In [51]:
subset_df = df[df['Class'].isin(['Fruits and Vegetables'])]
num_rows = len(subset_df)
print("Number of rows in subset_df:", num_rows)
countnull = subset_df.isnull().sum()
print(countnull)

Number of rows in subset_df: 9184
YearStart                        0
YearEnd                          0
LocationAbbr                     0
LocationDesc                     0
Datasource                       0
Class                            0
Topic                            0
Question                         0
Data_Value_Unit                  0
Data_Value_Type                  0
Data_Value                    1029
Data_Value_Alt                1029
Data_Value_Footnote_Symbol    8155
Data_Value_Footnote           8155
Low_Confidence_Limit          1029
High_Confidence_Limit         1029
Sample_Size                   1029
Total                         8856
Age(years)                    7216
Education                     7872
Gender                        8528
Income                        6888
Race/Ethnicity                6560
GeoLocation                    168
ClassID                          0
TopicID                          0
QuestionID                       0
DataValueTypeID      

These large amounts of null values are acually indended. This Dataset categorises not only by class and question, but also by statification categories

Number of null values in the Age column with the Stratification Category Age

In [52]:
subset2_df = df[df['StratificationCategory1'].isin(['Age (years)'])]
num_rows = len(subset2_df)
countnull = subset2_df.isnull().sum().sort_values()
print(countnull)
print("Number of rows in subset2_df:", num_rows)

YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
Datasource                        0
Class                             0
Topic                             0
Question                          0
Data_Value_Type                   0
TopicID                           0
QuestionID                        0
DataValueTypeID                   0
LocationID                        0
StratificationCategory1           0
Stratification1                   0
Age(years)                        0
StratificationCategoryId1         0
ClassID                           0
StratificationID1                 0
Data_Value                      240
High_Confidence_Limit           240
Data_Value_Alt                  240
Low_Confidence_Limit            240
Sample_Size                     240
GeoLocation                     414
Data_Value_Unit                3300
Data_Value_Footnote_Symbol    22104
Data_Value_Footnote         

as we can see, there are no null values in the Age column when sorting for the Stratification category Age

YearStart and YearEnd is alsways the same

In [56]:
sameyear = df['YearStart'] == df['YearEnd']
truecount = sameyear.sum()
falsecount = (~sameyear).sum()
print(truecount)
print(falsecount)

104272
0
